src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -136,6 +136,37 @@ class JcrNodeDecorator {
     }
 
     /**
+     * Finds out nearest versionable ancestor for a node and performs a checkout
+    */
+    void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor()
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path)
+            }
+        }
+        catch (Exception exception) {
+            log.warn "Could not checkout node ${this.path}, ${exception.message}"
+        }
+    }
+
+    private JcrNodeDecorator findVersionableAncestor() {
+        if (isVersionable()) {
+            return this
+        }
+        try {
+            JcrNodeDecorator parentDecoratedNode = new JcrNodeDecorator(this.parent)
+            return parentDecoratedNode.findVersionableAncestor()
+        } catch (ItemNotFoundException e) {
+            return null
+        }
+    }
+
+    private boolean isVersionable() {
+        return isNodeType(MIX_VERSIONABLE) || isNodeType("mix:simpleVersionable")
```

Comment:
Yeah.. so I did not use NodeType class to refer simple version... the one that you mentioned is referred from jcr constant that is shor value